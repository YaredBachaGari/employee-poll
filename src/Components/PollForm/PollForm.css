.pollform-container {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  text-align: center;
}
.pollform-input-container{
  width: 100%;
  margin: 0.25rem auto;
}
.pollform-input-container input {
  display: block;
  width: 90%;
  height: 2rem;
  margin: 0rem auto;
  border-radius: 0.5rem;
  border: 0.5px solid #737272;
  font-size: 1rem;
}
.pollform-btn button{
    font-size: 1rem;
    font-weight: bold;
    padding:0.5rem 2rem;
    margin-top: 1.5rem;
    background-color:#eeeeee;
    border: 1px solid #737272;
    border-radius: 0.5rem;
}

.pollform-btn button:hover{
    background-color:brown; 
    color: #eeeeee;
}
.formheading{
font-size: 1.5rem;
font-weight: bold;
}
.suggestion{
  font-size: 0.9rem;
  color: #737272;
}