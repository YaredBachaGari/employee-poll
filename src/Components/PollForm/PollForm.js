import React, { useState } from "react";
import "./PollForm.css";
import { handleQuestionPost } from "../../Redux-handler/Actions/Questions";
import { useSelector, useDispatch } from "react-redux";
import { useNavigate } from "react-router-dom";

const PollForm = () => {
  const dispatch = useDispatch();
  const navigate = useNavigate();
  const { loggedInUser } = useSelector((state) => state.AuthUser);
  const [optionsText, seOptionsText] = useState({
    optionOne: "",
    optionTwo: "",
    sucess: false,
  });
  const onChangeHandler = (e) => {
    const name = e.target.name;
    const value = e.target.value;
    seOptionsText((prev) => {
      return {
        ...prev,
        [name]: value,
      };
    });
  };

  const onSubmit = (e) => {
    e.preventDefault();
    const optionOneText = optionsText.optionOne;
    const optionTwoText = optionsText.optionTwo;
    const author = loggedInUser.username;
    const data = {
      optionOneText: optionOneText,
      optionTwoText: optionTwoText,
      author: author,
    };
    dispatch(handleQuestionPost(data));
    seOptionsText({
      optionOne: "",
      optionTwo: "",
      sucess: true,
    });
    navigate("/home");
  };
  return (
    <div className="pollform-container">
      {optionsText.sucess && (
        <p className="successMsg" data-testid="successmg">
          you have successfully created a poll!!
        </p>
      )}
      <div>
        <p className="formheading">Would You Rather</p>
        <p className="suggestion">Create Your Own Poll</p>
      </div>

      <div className="pollform-input-container">
        <label htmlFor="firstoption">First Option</label>
        <input
          id="firstoption"
          name="optionOne"
          data-testid="optionOne"
          value={optionsText.optionOne}
          placeholder="Option one"
          onChange={onChangeHandler}
        />
      </div>
      <div className="pollform-input-container">
        <label htmlFor="secondoption">Second Option</label>
        <input
          id="secondoption"
          name="optionTwo"
          data-testid="optionTwo"
          value={optionsText.optionTwo}
          placeholder="Option two"
          onChange={onChangeHandler}
        />
      </div>
      <div className="pollform-btn">
        <button
          className="btn"
          onClick={onSubmit}
          data-testid="createPoll"
          disabled={
            optionsText.optionOne && optionsText.optionTwo ? false : true
          }
        >
          Submit
        </button>
      </div>
    </div>
  );
};

export default PollForm;
