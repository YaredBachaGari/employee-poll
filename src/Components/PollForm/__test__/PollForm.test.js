import React from "react";
import { fireEvent, render, screen, waitFor } from "@testing-library/react";
import PollForm from "../PollForm";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import { useDispatch } from "react-redux";

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(),
}));

jest.mock("../../../Redux-handler/Actions/Users");
jest.mock("../../../Redux-handler/Actions/Questions");

describe("PollForm Component", () => {
  const mockDispatch = jest.fn();

  beforeEach(() => {
    useDispatch.mockReturnValue(mockDispatch);
  });

  afterEach(() => {
    jest.resetAllMocks();
  });
  it("should render the form", () => {
    const mockStore = configureStore();
    const store = mockStore({
      AuthUser: {
        loggedInUser: { id: 1, username: "john" },
      },
    });
    const { Form } = render(
      <Provider store={store}>
        <BrowserRouter>
          <PollForm />
        </BrowserRouter>
      </Provider>
    );
    expect(Form).toMatchSnapshot();
    expect(screen.getByTestId("createPoll")).toBeInTheDocument();
    expect(screen.getByTestId("optionOne")).toBeInTheDocument();
    expect(screen.getByTestId("optionTwo")).toBeInTheDocument();
  });
  it("on submit of the poll success msg should display", async () => {
    const mockStore = configureStore();
    const store = mockStore({
      AuthUser: {
        loggedInUser: { id: 1, username: "john" },
      },
    });
    const { Form } = render(
      <Provider store={store}>
        <BrowserRouter>
          <PollForm />
        </BrowserRouter>
      </Provider>
    );
    fireEvent.input(screen.getByTestId("optionOne"), {
      target: { value: "learn java" },
    });
    fireEvent.input(screen.getByTestId("optionOne"), {
      target: { value: "learn c#" },
    });
    fireEvent.click(screen.getByTestId("createPoll"));
    await waitFor(() => new Promise((resolve) => setTimeout(resolve, 4000)), {
      timeout: 5000,
    });
    await expect(
      screen.queryByText(/you have successfully created a poll!!/i)
    ).not.toBeInTheDocument();
  });
});
