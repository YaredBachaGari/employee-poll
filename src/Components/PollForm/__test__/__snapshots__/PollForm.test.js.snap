// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollForm Component should render the form 1`] = `undefined`;
